docs: expand documentation for secrets

lorenzo-cavazzi opened the pull request:
Documentation for the new "User secrets"

lokijuhy commented on docs/topic-guides/secrets/secrets.rst:
```diff
@@ -0,0 +1,138 @@
+reasons. You can still change it by clicking on the ``Replace`` button,
+or remove it by clicking on the ``Delete`` button. The name cannot be changed;
+should you need to rename a secret, please delete it and create a new one
+you the new name.
```
```suggestion
with the new name.
```

lokijuhy commented on docs/topic-guides/secrets/secrets.rst:
```diff
@@ -0,0 +1,138 @@
+Use secrets in sessions
+-----------------------
+
+If you need to include secrets in a session, you need to click on the Start
```
```suggestion
To use secrets in a session, you need to click on the Start
```

lokijuhy commented on docs/topic-guides/secrets/secrets.rst:
```diff
@@ -0,0 +1,138 @@
+dropdown menu and select ``Start with options``. Quick-start sessions do not
+support secrets.
+
+Once on the "Start Session" page, you can select the secrets you want to
```
```suggestion
Once on the "Start with options" page, you can select the secrets you want to
```

lokijuhy reviewed (changes requested):
Looks good! Just a few minor suggestions

rokroskar commented on docs/topic-guides/secrets/secrets.rst:
```diff
@@ -0,0 +1,138 @@
+  :align: center
+  :alt: Add a new secret
+
+Once you add a secret, you cannot visualize its value again for security
```
```suggestion
Once you add a secret, you cannot see its value again for security
```

rokroskar commented on docs/topic-guides/secrets/secrets.rst:
```diff
@@ -0,0 +1,138 @@
+``user key`` to undo the inner encryption. It then creates files inside the 
+session with the decrypted secret values.
+
+If your data is extra sensitive, consider putting already encrypted values into 
```
Should we add here a blurb along the lines of: 

```suggestion
Note that although we take many precautions to decrypt secrets only when necessary, they are in plaintext inside a session. This means that they are visible to whoever has access to the infrastructure where the session is running. If your data is extra sensitive, consider putting already encrypted values into 
```

rokroskar reviewed (changes requested):
Really nice docs, thanks for writing this up! Just a few minor comments and suggestions

sgaist commented on docs/spelling_wordlist.txt:
```diff
@@ -296,6 +297,7 @@ txt
 typesystem
 Ubuntu
 ui
+unencrypted
```
I would suggest to remove it and use `decrypted`.

sgaist commented on docs/topic-guides/secrets/secrets.rst:
```diff
@@ -0,0 +1,143 @@
+``Value`` fields.
+
+The name is a unique identifier for the secret, used for the file name in
+sessions. It cannot be empty and must follow validation rules:
```
```suggestion
sessions. It cannot be empty and must follow these validation rules:
```

sgaist commented on docs/topic-guides/secrets/secrets.rst:
```diff
@@ -0,0 +1,143 @@
+  :alt: Secrets encryption scheme
+
+Renku stores secrets in its database, doubly encrypted, ensuring that no part 
+accessible from the internet other than the session has access to unencrypted 
```
```suggestion
accessible from the internet other than the session has access to decrypted
```

sgaist commented on docs/topic-guides/secrets/secrets.rst:
```diff
@@ -0,0 +1,143 @@
+service`` that a user would like to start a session with some secret mounted. 
+It uses its ``private key`` to get the ``secret key`` and uses this to decrypt 
+the outer layer of encryption of the secret. It then creates a Kubernetes 
+secret with the (now singly encrypted) user secret, which gets mounted in the 
```
```suggestion
secret with the (now encrypted once) user secret, which gets mounted in the 
```

sgaist commented on docs/topic-guides/secrets/secrets.rst:
```diff
@@ -0,0 +1,143 @@
+On session start, an init container reads the mounted secrets, and uses the 
+``user key`` to undo the inner encryption. It then creates files inside the 
+session with the decrypted secret values.
+
```
May I nitpick a bit ?
The Kubernetes secret is in fact only mounted in the init container that will decrypt its content. The session (if we are talking about the main container) will only see the volume containing the decrypted secrets.

lorenzo-cavazzi replied:
I'll let @Panaetius answer this technical aspect :grin:

Panaetius replied:
I've changed the description a bit (I think this comment meant to address the previous paragraph, not this one)

sgaist reviewed (changes requested):
Overall it looks nice.

The encryption / decryption gif is excellent !

lorenzo-cavazzi commented:
@sgaist @rokroskar @lokijuhy Thank you for reading through this  :pray: 
I incorporated all your suggestions; please feel free to comment further should you find anything else.

@lokijuhy I added changes to the release notes here c187e4aa937efd14d660fa185623729a7a86ef79 . Mind that the link won't work until we publish the release since it's a new page.

lokijuhy reviewed (approved):
🎉
